Src/DependencyCollector/Microsoft.AI.DependencyCollector.Tests/DependencyCollectorDiagnosticListenerTests.cs

```diff
@@ -0,0 +1,251 @@
+
+            DependencyTelemetry telemetry = listener.PendingDependencyTelemetry.Single();
+            Assert.AreEqual("", telemetry.ResultCode);
+            Assert.AreEqual(true, telemetry.Success);
+
+            HttpResponseMessage response = new HttpResponseMessage(HttpStatusCode.OK);
+            string targetInstrumentationKeyHash = InstrumentationKeyHashLookupHelper.GetInstrumentationKeyHash(instrumentationKey);
+            response.Headers.Add(DependencyCollectorDiagnosticListener.TargetInstrumentationKeyHeader, targetInstrumentationKeyHash);
+
+            listener.OnResponse(response, loggingRequestId);
+            Assert.AreEqual(0, listener.PendingDependencyTelemetry.Count());
+            Assert.AreEqual(1, sentTelemetry.Count);
+            Assert.AreSame(telemetry, sentTelemetry.Single());
+
+            Assert.AreEqual("Http", telemetry.Type);
+            Assert.AreEqual("www.example.com", telemetry.Target);
+            Assert.AreEqual("200", telemetry.ResultCode);
+            Assert.AreEqual(true, telemetry.Success);
+        }
+
+        [TestMethod]
+        public void OnResponseWithFailedResponseEventWithMatchingRequestAndSameTargetInstrumentationKeyHashHeader()
+        {
+            Guid loggingRequestId = Guid.NewGuid();
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, "https://www.example.com");
+            listener.OnRequest(request, loggingRequestId);
+            Assert.AreEqual(1, listener.PendingDependencyTelemetry.Count());
+            Assert.AreEqual(0, sentTelemetry.Count);
+
+            DependencyTelemetry telemetry = listener.PendingDependencyTelemetry.Single();
+            Assert.AreEqual("", telemetry.ResultCode);
+            Assert.AreEqual(true, telemetry.Success);
+
+            HttpResponseMessage response = new HttpResponseMessage(HttpStatusCode.NotFound);
+            string targetInstrumentationKeyHash = InstrumentationKeyHashLookupHelper.GetInstrumentationKeyHash(instrumentationKey);
+            response.Headers.Add(DependencyCollectorDiagnosticListener.TargetInstrumentationKeyHeader, targetInstrumentationKeyHash);
+
+            listener.OnResponse(response, loggingRequestId);
+            Assert.AreEqual(0, listener.PendingDependencyTelemetry.Count());
+            Assert.AreEqual(1, sentTelemetry.Count);
+            Assert.AreSame(telemetry, sentTelemetry.Single());
+
+            Assert.AreEqual("Http", telemetry.Type);
+            Assert.AreEqual("www.example.com", telemetry.Target);
+            Assert.AreEqual("404", telemetry.ResultCode);
+            Assert.AreEqual(false, telemetry.Success);
+        }
+
+        [TestMethod]
+        public void OnResponseWithSuccessfulResponseEventWithMatchingRequestAndDifferentTargetInstrumentationKeyHashHeader()
+        {
+            Guid loggingRequestId = Guid.NewGuid();
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, "https://www.example.com");
```

Comment:
nitpick: it might be nice to make constants for the repeated magic strings (e.g. "https://www.example.com").